actions/telemetry_actions.jsx

```diff
@@ -92,22 +99,70 @@ export function measure(name1, name2) {
     return [lastDuration, measurementName];
 }
 
-export function trackLoadTime() {
+function trackPageLoadTimeEvent() {
     if (!isSupported([performance.timing.loadEventEnd, performance.timing.navigationStart])) {
         return;
     }
 
+    const {loadEventEnd, navigationStart} = window.performance.timing;
+    const pageLoadTime = loadEventEnd - navigationStart;
+    trackEvent('performance', 'page_load', {duration: pageLoadTime});
+}
+
+/**
+ * Measures the time and number of requests on first page load.
+ */
+export function measurePageLoadTelemetry() {
     // Must be wrapped in setTimeout because loadEventEnd property is 0
     // until onload is complete, also time added because analytics
     // code isn't loaded until a subsequent window event has fired.
     const tenSeconds = 10000;
     setTimeout(() => {
-        const {loadEventEnd, navigationStart} = window.performance.timing;
-        const pageLoadTime = loadEventEnd - navigationStart;
-        trackEvent('performance', 'page_load', {duration: pageLoadTime});
+        trackPageLoadTimeEvent();
+        trackNumOfRequestEvent('page_load');
     }, tenSeconds);
 }
 
+/**
+ * Clears all performance entries of entry type "resource".
+ */
+export function clearResourceTimings() {
+    if (!shouldTrackPerformance() || !SUPPORTS_RESOURCE_TIMINGS) {
+        return;
+    }
+
+    performance.clearResourceTimings();
+}
+
+/**
+ * Send an telemetry for number of requests to the server for an event.
+ * @param {String} initiatedBy The name of the event that initiated the tracking
+ */
+export function trackNumOfRequestEvent(initiatedBy) {
```

Comment:
Only concern with this is that it's not tied to the "initiatedBy". So if this function is called in two different places around the same time the metrics could get all messed up for. For example, if immediately after page load someone does a channel switch we might report all the requests as part of the channel switch if the page load timeout hasn't fired yet. Maybe when we "mark" the start of an event like `page_load` we only let `trackNumOfRequestEvent` run when it's called with `page_load`. Essentially a lock on the function. Not sure if it's necessary or overkill. What do you think?

Reply:
Instead of clearing the timings between calls for this, if we tied this to the page_load/channel_switch event, we could count only resource entries that occurred within the start and end of that event by using the `startTime` and `duration`. That should allow us to count requests for two events in parallel